app/src/main/java/woowacourse/shopping/view/main/MainActivity.kt

```
@@ -126,32 +114,49 @@ class MainActivity : AppCompatActivity() {
+
+    private fun setSpanSizeLookup() =
+        object : GridLayoutManager.SpanSizeLookup() {
+            override fun getSpanSize(position: Int): Int {
+                return when (productsAdapter.getItemViewType(position)) {
+                    ProductRvItems.ViewType.VIEW_TYPE_PRODUCT.ordinal -> 1
+                    ProductRvItems.ViewType.VIEW_TYPE_RECENT_PRODUCT.ordinal,
+                    ProductRvItems.ViewType.VIEW_TYPE_DIVIDER.ordinal,
+                    ProductRvItems.ViewType.VIEW_TYPE_LOAD.ordinal,
+                    -> 2
+
+                    else -> throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)